build/utils-node.js

```diff
@@ -10,179 +10,540 @@
-    } else {
-      cmds = cmd + ' ' + cmds;
-    }
-    console.log(cmds);
-    // XXX: Most cmds should run synchronously, we should use either promise
-    //      pattern inside each script or find a sync module which doesn't
-    //      require recompile again since TPBL doesn't support that.
-    sh(cmds, function(err, stdout, stderr) {
-      if (err === null) {
-        callback && callback(0);
+  joinPath: function() {
+    var src = path.join.apply(path, arguments);
+    return src;
+  },
+
+  getOsType: function() {
+    var type = os.type();
+    return type === 'Windows_NT' ? 'WINNT' : type;
+  },
+
+  relativePath: function(from, to) {
+    return path.relative(from, to);
+  },
+
+  getFile: function() {
+    var src = path.join.apply(path, arguments);
+    var fileStat;
+    try {
+      fileStat = fs.statSync(src);
+    } catch (e) {
+      // In order to have compability with xpcshell interface,
+      // we just simply let fileStat undefined and don't need
+      // to do anything here.
+    }
+    return {
+      exists: function() {
+        return fs.existsSync(src);
+      },
+      remove: function() {
+        fs.remove(src);
+      },
+      isDirectory: function() {
+        return !!fileStat && fileStat.isDirectory();
+      },
+      isFile: function() {
+        return !!fileStat && fileStat.isFile();
+      },
+      isHidden: function() {
+        return /^\./.test(path.basename(src));
+      },
+      path: src,
+      leafName: path.basename(src)
+    };
+  },
+
+  getFileContent: function(file, type) {
+    if (file.exists() && file.isFile()) {
+      return fs.readFileSync(file.path, { encoding: type || 'utf8' });
+    }
+  },
+
+  Commander: function(cmd) {
+    this.initPath = function(p) {};
+    this.run = function(args, callback) {
+      var q = Q.defer();
+      var cmds = args.join(' ');
+
+      // In *nix and OSX version commands are run via sh -c YOUR_COMMAND,
+      // but in Windows commands are run via cmd /C YOUR_COMMAND,
+      // so, we just let execSync module to handle the difference.
+      if (cmd === 'sh') {
+        cmds = cmds.replace(/^-c/, '');
       } else {
-        callback && callback(1);
+        cmds = cmd + ' ' + cmds;
+      }
+      console.log(cmds);
+      // XXX: Most cmds should run synchronously, we should use either promise
+      //      pattern inside each script or find a sync module which doesn't
+      //      require recompile again since TPBL doesn't support that.
+      childProcess.exec(cmds, function(err, stdout, stderr) {
+        if (err === null && typeof callback === 'function') {
+          callback(stdout);
+        }
+        q.resolve();
+      });
+      return q.promise;
+    };
+  },
+
+  readZipManifest: function(file) {
+    var zipPath = this.joinPath(file.path, 'application.zip');
+    if (!fs.existsSync(zipPath)) {
+      return;
+    }
+    var zip = new JSZip(fs.readFileSync(zipPath));
+    return zip.file('manifest.webapp').asText();
```

Comment:
According JSZip doc:  An instance of ZipObject representing the file if any, null otherwise.
I will fix it soon.